chapter04/chapter04.adoc

```diff
@@ -0,0 +1,317 @@
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import java.util.List;
+
+@Path("/products")
+public class ProductResource {
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "List all products", description = "Retrieves a list of all available products")
+    @APIResponses(value = {
+        @APIResponse(
+            responseCode = "200",
+            description = "Successful, list of products found",
+            content = @Content(mediaType = "application/json",
+                               schema = @Schema(implementation = Product.class, type = SchemaType.ARRAY))
+        ),
+        @APIResponse(
+            responseCode = "400",
+            description = "Unsuccessful, no products found",
+            content = @Content(mediaType = "application/json")
+        )
+    })
+    public List<Product> getProducts() {
+        // Method implementation
+    }
+}
+----
+
+Explanation: 
+
+* `@Operation`: Provides a summary and description for the `getProducts()` method.
+
+* `@APIResponse`: Describes the possible responses from the `getProducts()` operation. In this case, a successful response (HTTP 200) is described, indicating that the method returns an array of Product entities.
+
+* `@Schema`: Specifies the schema of the response content. Here, it is used to indicate that the method returns an array of Product objects.
+
+These annotations enrich the `ProductResource` class with metadata necessary for generating comprehensive and descriptive OpenAPI documentation automatically.
+
+We have also annotated the `getProducts()` method with the @APIResponse annotation to document the successful response from the operation. The `responseCode` field is used to specify the status code of the response, and the `description` field is used to provide a brief description of the response. There are two possible responses – a successful response containing a list of produdts with a 200 status code, and an unsuccessful response with a 400 status code, if no products are found. The content field is used to specify the schema of the response content. In this example, the response content is a list of MyObjects.
+
+Finally, we need to add the following property to the src/main/resources/META-INF/microprofile-config.properties file:
+
+----
+mp.openapi.scan=true
+----
+
+This property tells MicroProfile OpenAPI to scan our classes for annotations and generate API documentation for them. 
+
+Now that we have configured MicroProfile OpenAPI, we can build and run our application.
+
+=== How to view the generated documentation
+
+To view the generated documentation, we can use the OpenAPI UI tool. The Open API UI tool is a web-based tool that can be used to view the documentation for a REST API.
+
+The OpenAPI UI tool can be accessed at the following URL:
+
+----
+http://localhost:8080/openapi/
```

Comment:
Just to note, the port isn't part of the specification and may vary depending on the server/runtime being used.

Reply:
Fixed with https://github.com/ttelang/microprofile-tutorial/commit/2e7a527447747d338c4748f088218dee0ddd4056